app/controllers/save_pages_controller.rb

```diff
@@ -0,0 +1,45 @@
+class SavePagesController < ApplicationController
+  include AccountConcern
+
+  before_action :set_no_cache_headers
+  before_action { head :not_found unless feature_flag_enabled? }
+  before_action :authenticate
+
+  def create
+    GdsApi.account_api.save_page(
+      page_path: ensure_is_path(params[:page_path]),
+      govuk_account_session: account_session_header,
+    )
+    redirect_to ensure_is_path(params[:page_path]) + "?personalisation=page_saved"
+  rescue GdsApi::HTTPUnauthorized
+    @account_session_header = nil
+    authenticate
+  rescue GdsApi::HTTPUnprocessableEntity
+    redirect_to ensure_is_path(params[:page_path]) + "?personalisation=page_not_saved"
```

Comment:
Since there are three instances of `ensure_is_path(params[:page_path])`, how about sticking it in a variable?

Reply:
Do you mean like as a secnd param?

Like `ensure_is_path(params[:page_path], query_params)`

Where query params is a hash or something like:
```
{
   personalisation: "page_saved"
}
```

(I think addressable can do a better query param merge?)

Reply:
No, like

```ruby
  def create
    page_path = ensure_is_path(params[:page_path])
    GdsApi.account_api.save_page(
      page_path: page_path,
      ...
```

Reply:
Ok I see, sure.